Bottom: c744a2dc55ff0d7b35d1ce75bf5e4573290e6ef1
Top:    08098ee53015089143d1be527cbd454f1af9d1b3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-05-14 16:36:29 +0100

Refresh of fixup-line-rubber-band-code-to

---

diff --git a/src/crosshair.c b/src/crosshair.c
index 274ce69..b2a596f 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -438,10 +438,11 @@ XORDrawMoveOrCopyObject (hidGC gc)
         i = Crosshair.AttachedObject.RubberbandN;
         ptr = Crosshair.AttachedObject.Rubberband;
 
-        while (i)
+        for (; i; ptr++, i--)
           {
             if (!TEST_FLAG (VIAFLAG, ptr->Line) &&
-                TEST_FLAG (RUBBERENDFLAG, ptr->Line))
+                TEST_FLAG (RUBBERENDFLAG, ptr->Line) &&
+                ptr->Layer != NULL)
               {
                 double multiplier;
 
@@ -471,9 +472,6 @@ XORDrawMoveOrCopyObject (hidGC gc)
                 min_multiplier = MIN (min_multiplier, multiplier);
                 max_multiplier = MAX (max_multiplier, multiplier);
               }
-
-            ptr++;
-            i--;
           }
 
 
@@ -597,16 +595,15 @@ XORDrawMoveOrCopyObject (hidGC gc)
   i = Crosshair.AttachedObject.RubberbandN;
   ptr = Crosshair.AttachedObject.Rubberband;
 //  while (i)
-  while (i)
+  for (; i; ptr++, i--)
     {
       PointType *point1, *point2;
 
+      /* this is a rat going to a polygon.  do not draw for rubberband */;
       if (TEST_FLAG (VIAFLAG, ptr->Line))
-	{
-	  /* this is a rat going to a polygon.  do not draw for rubberband */;
-	}
-      else if (TEST_FLAG (RUBBERENDFLAG, ptr->Line))
-//      else
+        continue;
+
+      if (TEST_FLAG (RUBBERENDFLAG, ptr->Line))
 	{
           if (Crosshair.AttachedObject.Type == LINE_TYPE)
             {
@@ -658,9 +655,6 @@ XORDrawMoveOrCopyObject (hidGC gc)
                              ptr->Line->Point2.X + dx,
                              ptr->Line->Point2.Y + dy, ptr->Line->Thickness);
 #endif
-
-      ptr++;
-      i--;
     }
 }
 
diff --git a/src/move.c b/src/move.c
index 6282e9e..cfdf881 100644
--- a/src/move.c
+++ b/src/move.c
@@ -874,7 +874,8 @@ MoveObjectAndRubberband (int Type, void *Ptr1, void *Ptr2, void *Ptr3,
   ptr = Crosshair.AttachedObject.Rubberband;
   while (Crosshair.AttachedObject.RubberbandN)
     {
-      if (Type == LINE_TYPE)
+      if (Type == LINE_TYPE &&
+          ptr->Layer != NULL)
         {
           double x, y, multiplier;
